workbench_fields.py

```diff
@@ -1255,6 +1255,205 @@ def serialize(self, config, field_definitions, field_name, field_data):
         else:
             return subvalues[0]
 
+class TypedRelationDisplayNameField(TypedRelationField):
+    """Functions for handling fields with 'typed_relation_display_name' Drupal field data type.
+    All functions return an "entity" dictionary that is passed to Requests' "json"
+    parameter.
+
+    Currently this field type only supports Typed Relation Taxonomies (not other
+    Typed Relation entity types).
+
+    Note: this class assumes that the entity has the field identified in 'field_name'.
+    Callers should pre-emptively confirm that. For an example, see code near the top
+    of workbench.update().
+    """
+
+    def create(self, config, field_definitions, entity, row, field_name):
+        """Parameters
+        ----------
+         config : dict
+             The configuration settings defined by workbench_config.get_config().
+         field_definitions : dict
+             The field definitions object defined by get_field_definitions().
+         entity : dict
+             The dict that will be POSTed to Drupal as JSON.
+         row : OrderedDict.
+             The current CSV record.
+         field_name : string
+             The Drupal fieldname/CSV column header.
+         Returns
+         -------
+         dictionary
+             A dictionary representing the entity that is POSTed to Drupal as JSON.
+        """
+        if not row[field_name]:
+            return entity
+
+        id_field = row.get(config.get("id_field", "not_applicable"), "not_applicable")
+        # Currently only supports Typed Relation Display Name with taxonomy entities.
+        if field_definitions[field_name]["target_type"] == "taxonomy_term":
+            target_type = "taxonomy_term"
+            field_vocabs = get_field_vocabularies(config, field_definitions, field_name)
+            field_values = []
+            subvalues = split_typed_relation_display_name_string(
```

Comment:
I tried to make this class as short as possible as it inherits from TypedRelationField. This Class could be much shorter if we made the "split into subvalues" function a parameter of the class. In the create() and update() functions, the ONLY difference is using my custom "split_typed_relation_display_name_string" function.